File: 2515-shortest-distance-to-target-string-in-a-circular-array/2515-shortest-distance-to-target-string-in-a-circular-array.cpp
class Solution {
public:
    int closetTarget(vector<string>& words, string target, int startIndex) {
        
        
        int n = words.size();
        
        //simple thing to maintain the visited node when its pass that mean its visited true if all the node visited than 
        //terminated the loop 
        vector<bool>visited(n,false);
        //left move         // right move 
        int l = startIndex,r = startIndex;
        int res = 0;
        
        while(!visited[l] || !visited[r]){
            
            if(words[l] == target || words[r] == target){
                return res;
            }
            
            visited[l] = true;
            visited[r] = true;
            l--;
            r++;
            
            res++;
            
            //when l value -1 that means follow to circular move that means going to last index 
            l = l == -1 ? n-1:l;
            
            //its also when right pointer come to the n/ isequal to n than its move forward starting pointer
            r = r == n ? 0 : r;
        }
        
        return -1;
        
    }
};

File: 2515-shortest-distance-to-target-string-in-a-circular-array/2515-shortest-distance-to-target-string-in-a-circular-array.js
/**
 * @param {string[]} words
 * @param {string} target
 * @param {number} startIndex
 * @return {number}
 */
var closetTarget = function(words, target, startIndex) {
    
 
   let l = startIndex,r = startIndex;
    let res = 0;
    let n = words.length;
    let visited = new Array(n).fill(false);
    console.log(visited)
    while(!visited[l] && !visited[r]){
        if(words[l] === target || words[r] === target){
            return res;
        }
        visited[l] = true;
        visited[r] = true;
        l--;
        r++;
        res++;
        l = l == -1 ? n - 1 : l;
        r = r == n ? 0 : r;
    }
    return -1;
};